### DISCUSSION TITLE
How does StreamingResponse work?

### DISCUSSION THREAD

**Discussion by jamesbraza** [fastapi/fastapi#10761] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi.responses import StreamingResponse
```

### Description

How does `StreamingResponse` work? Is it using WebHooks, WebSockets, HTTP streaming, etc?

Coming from:
- https://fastapi.tiangolo.com/advanced/custom-response/#streamingresponse
- https://medium.com/llm-projects/langchain-openai-streaming-101-in-python-edd60e84c9ca

### Operating System

macOS

### Operating System Details

Darwin N7L493PWK4 22.6.0 Darwin Kernel Version 22.6.0: Wed Jul  5 22:22:05 PDT 2023; root:xnu-8796.141.3~6/RELEASE_ARM64_T6000 arm64

### FastAPI Version

0.104.1

### Pydantic Version

2.5.2

### Python Version

3.11.5

**Comment by cirezd** [comment]
If you want to stream chunks of tokens/text (you mentioned langchain), then have a look at https://github.com/sysid/sse-starlette. This provides a response class that handles ServerSentEvents (https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events). The streaming of tokens via SSE is also used by ChatGPT.

**Reply by ArijitSinghEDA** [reply]
Can you provide an example using FastAPI? I am assuming `StreamingResponse` gives output on newline character `\n`, so it does not feel as smooth as ChatGPT, for example.

**Comment by jamesbraza** [comment]
Made https://github.com/tiangolo/fastapi/pull/10932 to upstream some knowledge
